include/rapidjson/rapidjson.h

```diff
@@ -591,6 +591,19 @@ RAPIDJSON_NAMESPACE_END
 
 //!@endcond
 
+///////////////////////////////////////////////////////////////////////////////
+// RAPIDJSON_NOEXCEPT_ASSERT
+
+#ifdef RAPIDJSON_ASSERT_THROWS
+#if RAPIDJSON_HAS_CXX11_NOEXCEPT
+#define RAPIDJSON_NOEXCEPT_ASSERT(x)
```

Comment:
Removing the assertion entirely is not the ideal solution.  I would prefer to 
* allow a separate override of `RAPIDJSON_NOEXCEPT_ASSERT`
* defaulting to 
  * `RAPIDJSON_ASSERT`, if `!RAPIDJSON_HAS_CXX11_NOEXCEPT`
  * plain `assert` (i.e. default of `RAPIDJSON_ASSERT`) otherwise

Something like:
````cpp
//! Assertion (in non-throwing contexts).
 /*! \ingroup RAPIDJSON_CONFIG
    Some functions provide a \c noexcept guarantee, if the compiler supports it.
    In these cases, the \ref RAPIDJSON_ASSERT macro cannot be overridden to
    throw an exception.  This macro adds a separate customization point for
    these cases.

    Defaults to C \c assert() (as \ref RAPIDJSON_ASSERT), if \c noexcept is
    supported, and to \ref RAPIDJSON_ASSERT otherwise.
 */
#ifndef RAPIDJSON_NOEXCEPT_ASSERT
#if RAPIDJSON_HAS_CXX11_NOEXCEPT
#include <cassert>
#define RAPIDJSON_NOEXCEPT_ASSERT(x) assert(x)
#else
#define RAPIDJSON_NOEXCEPT_ASSERT(x) RAPIDJSON_ASSERT(x)
#endif // RAPIDJSON_HAS_CXX11_NOEXCEPT
#endif // RAPIDJSON_NOEXCEPT_ASSERT
````